crates/oxc_linter/src/lib.rs

```diff
@@ -92,7 +109,14 @@ impl Linter {
         Self::from_rules(rules)
     }
 
-    pub fn run<'a>(&self, ctx: LintContext<'a>) -> Vec<Message<'a>> {
+    /// relative_file_path_parts should be a vec of path parts relative to the file being linted
+    /// ie: if the linter is called from /, and the linted file is /apple/index.ts, then relative_file_path_parts is vec!["apple", "index.ts"]
+    pub fn run<'a>(
+        &self,
+        ctx: LintContext<'a>,
+        // The string is optional because some OsString's may not be represented as String's
+        relative_file_path_parts: Vec<Option<String>>,
```

Comment:
breaking change!

Reply:
Let's add this to `LintContext`, probably in a PR merged in main first?

Why is this not a `PathBuf` btw?

Reply:
Easier to deal with and test a `Vec<Option<String>>`, and that's what we work with anyway, we don't want to deal with path seperators so we exclusively use path components in queries, which are essentially `Vec<Option<String>>`, Option because some path components aren't String compatible

Reply:
Yeah, we can merge this first

Reply:
No longer necessary